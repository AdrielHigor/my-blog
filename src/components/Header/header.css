header#main-header {
  background-color: #25282a;
}

header#main-header .header-content{
    display: flex;
    width: 100%;
    height: 72px;
    margin: 0 auto;
    padding: 0 30px;
    align-items: center;
    justify-content: space-between;
    overflow: hidden;
}

header#main-header .header-content a{
    float: left;
    color: #ffffff;
    text-align: center;
    padding: 12px;
    text-decoration: none;
    font-size: 18px;
    line-height: 25px;
    border-radius: 4px
}