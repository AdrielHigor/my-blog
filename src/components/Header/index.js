import React, { useEffect, useState } from "react";
import { Link, withRouter } from "react-router-dom";
import "./header.css";
import firebase from "../../firebase";

export default withRouter(function Header(props) {
  const [data, setData] = useState({ link: "login", text: "Entrar" });

  useEffect(() => {
    function changeHeader() {
      if (firebase.getCurrent()) {
        setData({
          link: "dashboard",
          text: "Dashboard"
        });
      } else {
        setData({
          link: "login",
          text: "Entrar"
        });
      }
    }

    props.history.listen((location, action) => {
      if (action === "REPLACE") {
        changeHeader();
      }
    });

    changeHeader();
  }, []);

  return (
    <header id="main-header">
      <div className="header-content">
        <Link to="/">Meu Blog</Link>
        <Link to={"/" + data.link}>{data.text}</Link>
      </div>
    </header>
  );
});
