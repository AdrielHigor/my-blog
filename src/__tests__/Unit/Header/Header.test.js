import React from "react";
import Header from "../../../components/Header";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from "enzyme";

configure({ adapter: new Adapter() });

describe("Header Test Suite", () => {
    let component

    beforeEach(() => {
        component = shallow(<Header.WrappedComponent />)
    })

    test("Header rendering correctly", () => {
        expect(component).toMatchSnapshot();
    })
})