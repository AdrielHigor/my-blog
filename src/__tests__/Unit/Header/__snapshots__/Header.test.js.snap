// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test Suite Header rendering correctly 1`] = `ShallowWrapper {}`;
